RevenueCat/Scripts/Purchases.cs

```diff
@@ -1592,6 +1651,104 @@ private void _redeemWebPurchase(string redeemWebPurchaseJSON)
+        }
+        else
+        {
+            var winBackOffers = new List<WinBackOffer>();
+            foreach (JSONNode offerResponse in response["eligibleWinBackOffers"])
+            {
+                var offer = new WinBackOffer(offerResponse);
+                winBackOffers.Add(offer);
+            }
+
+            GetEligibleWinBackOffersForProductCallback(winBackOffers.ToArray(), null);
+        }
+
+        GetEligibleWinBackOffersForProductCallback = null;
+    }
+
+    private void _getEligibleWinBackOffersForPackage(string eligibleWinBackOffersJson)
+    {
+        Debug.Log("_getEligibleWinBackOffersForPackage " + eligibleWinBackOffersJson);
+
+        if (GetEligibleWinBackOffersForPackageCallback == null) return;
+
+        var response = JSON.Parse(eligibleWinBackOffersJson);
+
+        if (ResponseHasError(response))
+        {
+            GetEligibleWinBackOffersForPackageCallback(null, new Error(response["error"]));
+        }
+        else
+        {
+            var winBackOffers = new List<WinBackOffer>();
+            foreach (JSONNode offerResponse in response["eligibleWinBackOffers"])
+            {
+                var offer = new WinBackOffer(offerResponse);
+                winBackOffers.Add(offer);
+            }
+
+            GetEligibleWinBackOffersForPackageCallback(winBackOffers.ToArray(), null);
+        }
+
+        GetEligibleWinBackOffersForPackageCallback = null;
+    }
+
+    private void _purchaseProductWithWinBackOffer(string purchaseProductWithWinBackOfferJson)
+    {
+        Debug.Log("_purchaseProductWithWinBackOffer " + purchaseProductWithWinBackOfferJson);
+
+        if (MakePurchaseCallback == null) return;
+
+        var response = JSON.Parse(purchaseProductWithWinBackOfferJson);
+
+        if (ResponseHasError(response))
+        {
+            MakePurchaseCallback(null, null, response["userCancelled"], new Error(response["error"]));
+        }
+        else
+        {
+            var info = new CustomerInfo(response["customerInfo"]);
+            var productIdentifier = response["productIdentifier"];
+            MakePurchaseCallback(productIdentifier, info, false, null);
+        }
+
+        MakePurchaseCallback = null;
+    }
+
+    private void _purchasePackageWithWinBackOffer(string purchasePackageWithWinBackOfferJson)
+    {
+        Debug.Log("_purchasePackageWithWinBackOffer " + purchasePackageWithWinBackOfferJson);
+
+        if (MakePurchaseCallback == null) return;
+
+        var response = JSON.Parse(purchasePackageWithWinBackOfferJson);
+
+        if (ResponseHasError(response))
+        {
+            MakePurchaseCallback(null, null, response["userCancelled"], new Error(response["error"]));
+        }
+        else
+        {
+            var info = new CustomerInfo(response["customerInfo"]);
+            var packageIdentifier = response["packageIdentifier"];
+            MakePurchaseCallback(packageIdentifier, info, false, null);
```

Comment:
The `MakePurchaseCallback` first parameter is a `productIdentifier`, and you're passing the `packageIdentifier` here. I was looking at the implementation of PHC and it looks like in both `purchasePackage` and `purchaseProduct` we send the `productIdentifier` (you should double check but that's what I understand), so I think this should be sending the productIdentifier instead

Reply:
Great catch!! 💜

Reply:
Fixed in e4a03d765b7ccbebb24253aa031ebbfaf53b21d0!